qualtran/bloqs/arithmetic/addition.py

```diff
@@ -530,8 +530,8 @@ def get_ctrl_system(
         def _add_ctrled(
             bb: 'BloqBuilder', ctrl_soqs: Sequence['SoquetT'], in_soqs: Dict[str, 'SoquetT']
         ) -> Tuple[Iterable['SoquetT'], Iterable['SoquetT']]:
-            ctrl, x = bb.add_t(bloq, ctrls=ctrl_soqs[0], **in_soqs)
-            return (ctrl,), (x,)
+            ctrls, x = bb.add_t(bloq, ctrls=np.asarray(ctrl_soqs), **in_soqs)
+            return np.asarray(ctrls).tolist(), (x,)
```

Comment:
was this a bug? is there a test that would have caught it? 

actually: what's going on here? the conditions above exclude any situation in where there is more than one `SoquetT` being passed in. For this bloq, that one `SoquetT` is an array of soquets

Reply:
It fails on `test_tensor`:
```
            for li in reg.all_idxs():
>               idxed_soq = in_soq[li]
E               IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed
```